apps/aeprimop/src/aeprimop.erl

```diff
@@ -856,23 +862,52 @@ name_transfer({OwnerPubkey, RecipientType, RecipientHash, NameHash}, S) ->
 
 %%%-------------------------------------------------------------------
 
-name_update_op(OwnerPubkey, NameHash, DeltaTTL, MaxTTL, ClientTTL, Pointers
+name_update_op(OwnerPubkey, NameHash, DeltaTTL, ClientTTL, Pointers
               ) when ?IS_HASH(OwnerPubkey),
                      ?IS_HASH(NameHash),
                      ?IS_NON_NEG_INTEGER(DeltaTTL),
-                     ?IS_NON_NEG_INTEGER(MaxTTL),
                      ?IS_NON_NEG_INTEGER(ClientTTL),
                      is_list(Pointers) ->
-    {name_update , {OwnerPubkey, NameHash, DeltaTTL, MaxTTL, ClientTTL, Pointers}}.
-
-name_update({OwnerPubkey, NameHash, DeltaTTL, MaxTTL, ClientTTL, Pointers}, S) ->
-    [runtime_error(ttl_too_high) || DeltaTTL > MaxTTL],
-    {Name, S1} = get_name(NameHash, S),
-    assert_name_owner(Name, OwnerPubkey),
-    assert_name_claimed(Name),
-    AbsoluteTTL = S#state.height + DeltaTTL,
-    Name1 = aens_names:update(Name, AbsoluteTTL, ClientTTL, Pointers),
-    put_name(Name1, S1).
+    {name_update,
+     {OwnerPubkey, NameHash, {relative_ttl, DeltaTTL}, {relative_ttl, ClientTTL}, Pointers}};
+name_update_op(OwnerPubkey, NameHash, TTL, ClientTTL, Pointers
+              ) when ?IS_HASH(OwnerPubkey),
+                     ?IS_HASH(NameHash),
+                     ?IS_CHAIN_TTL_OPTION(TTL),
+                     ?IS_CHAIN_TTL_OPTION(ClientTTL),
+                     (Pointers == undefined orelse is_list(Pointers)) ->
+    {name_update, {OwnerPubkey, NameHash, TTL, ClientTTL, Pointers}}.
+
+
+name_update({OwnerPubkey, NameHash, TTL, ClientTTL, Pointers}, S) ->
```

Comment:
This will work, but I wonder if it would not be nicer to split `aens_names:update` into `aens_names:update_ttl`, `aens_names:update_pointers` and `aens_names:update_client_ttl`... And only update those values that are **not** `undefined` and thus not a have to dig out the old values...

Reply:
The usage from Sophia looks like this:

```
  stateful entrypoint update(owner      : address,
                             name       : string,
                             ttl        : option(Chain.ttl),
                             client_ttl : option(Chain.ttl),
                             pointers   : option(map(string, AENS.pointee))) : unit =
    AENS.update(owner, name, ttl, client_ttl, pointers)
```
So it's possible to update only what's needed without digging values just to be passed around. 

Agree on the ergonomics though, that ideally we would have all variants. (maybe at 5.3.0? it would all call the same thing at FATE level)